Update admission_controller.md (#15763)
DataDog/documentation: content/en/tracing/trace_collection/admission_controller.md

First, configure the Datadog Agent. Our recommended method of installation for Kubernetes environments is via Helm Charts. Learn how to configure the [Datadog Agent in Kubernetes via Helm Charts][3] by reading our documentation.

## Unified Service Tagging

> For your Kubernetes applications that send traces to Datadog Agent, you can configure the Datadog admission controller to inject Java, JavaScript and Python libraries automatically. To inject the Java and Javascript libraries, download Datadog Agent v7.39+. To inject the Python library, download Datadog Agent v7.40+.

Reviewer: ```suggestion
For your Kubernetes applications that send traces to Datadog Agent, you can configure the Datadog admission controller to inject Java, JavaScript and Python libraries automatically. To inject the Java and Javascript libraries, use Datadog Cluster Agent v7.39+. To inject the Python library, use Datadog Cluster Agent v7.40+.
```

Reviewer: Nit as the customer don't download the agent manually, they set the agent version in their yaml/values or just use the default one in the helm chart.

Reviewer 2: +1

Reviewer 2: also add a comma "... to inject Java, JavaScript, and Python..."

> For your Kubernetes applications that send traces to Datadog Agent, you can configure the Datadog admission controller to inject Java, JavaScript and Python libraries automatically. To inject the Java and Javascript libraries, use Datadog Cluster Agent v7.39+. To inject the Python library, use Datadog Cluster Agent v7.40+.

Reviewer 2: ```suggestion
For your Kubernetes applications that send traces to Datadog Agent, you can configure the Datadog admission controller to inject Java, JavaScript, and Python libraries automatically. To inject the Java and Javascript libraries, use Datadog Cluster Agent v7.39+. To inject the Python library, use Datadog Cluster Agent v7.40+.
```

    admission.datadoghq.com/js-lib.version: "v3.6.0"
```

> You can also inject the latest version of a tracer library for the Java and Python language via the following annotation:

Reviewer: Do we want to remove the node example on purpose?

> You can also inject the latest version of a tracer library for the Java and Python language via the following annotation:

Reviewer 2: ```suggestion
You can also inject the latest version of a tracer library using the following annotation:
```

You can also inject the latest version of a tracer library for the Java and Python language via the following annotation:
```yaml
annotations:
    admission.datadoghq.com/java-lib.version: "latest"

>     admission.datadoghq.com/js-lib.version: "latest"

Reviewer: same question

>     admission.datadoghq.com/python-lib.version: "latest"

Reviewer 2: ```suggestion
    admission.datadoghq.com/js-lib.version: "latest"
    admission.datadoghq.com/python-lib.version: "latest"
```

**Note**: Use caution specifying `latest`, as it may raise errors because of major version differences between your newly injected applications and your existing applications that have already been deployed. 

> Adding a `mutateUnlabelled: true` agent config in the helm chart causes the cluster agent to attempt to intercept every unlabelled pod.

Reviewer 2: ```suggestion
Adding a `mutateUnlabelled: true` Agent config in the Helm chart causes the Cluster Agent to attempt to intercept every unlabelled pod.
```

| `false`          | `admission.datadoghq.com/enabled=true`  | Yes       |
| `false`          | `admission.datadoghq.com/enabled=false` | No        |

## Configure the Tracer from a Different Registry

> You may pyll the APM library from a different registry using the `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_CONTAINER_REGISTRY` environment variable in the Datadog Cluster Agent.

Reviewer 2: ```suggestion
You may pull the APM library from a different registry using the `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_CONTAINER_REGISTRY` environment variable in the Datadog Cluster Agent.
```

> You may configure the tracer from a different registry using the `admission_controller.auto_instrumentation.container_registry` config in the cluster agent or the `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_CONTAINER_REGISTRY` environment variable.

Reviewer: ```suggestion
You may pyll the APM library from a different registry using the `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_CONTAINER_REGISTRY` environment variable in the Datadog Cluster Agent.
```

In containerized environments environment variables are recommended to configure the agent, as it's easier to add them instead of changing and mouting a datadog.yaml file.

Reviewer 2: +1, but "pull" not "pyll" (unless pyll is some python thing I was previously unaware of)

Reviewer: Yes sorry that's a typo 🤦 it's pull.

> The default value is set to `grc.io/datadoghq`. Any custom value must follow the same format and refer to a valid/existing registry.

Reviewer: ```suggestion
The default value is set to `grc.io/datadoghq`. The supported values are `docker.io/datadog` and `public.ecr.aws/datadog`.
```
